pkg/splunk/enterprise/util.go

```diff
@@ -331,8 +331,8 @@ func ApplyAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.Me
 
 	appListingConfigMap.SetOwnerReferences(append(appListingConfigMap.GetOwnerReferences(), splcommon.AsOwner(cr, true)))
 
-	if len(appListingConfigMap.Data) > 0 {
-		configMapDataChanged, err = splctrl.ApplyConfigMap(client, appListingConfigMap)
+	if len(appListingConfigMap.Data) > 0 || appsModified {
+		configMapDataChanged, err = splctrl.ApplyConfigMap(client, appListingConfigMap, appsModified)
```

Comment:
Since it is a unique situation for the Phase-23 app framework, i feel it m

Reply:
Ya this is only applicable for as long as we use the configmap method.  But its needed for the case were the set of applications that has been modified is the same as the previous change (i.e. no change in the configmap data section).

Reply:
My bad, seems like I messed up my comment. What I wanted to mention was, since it is a unique case just for phase-2, we can simply delete the existing configMap and then write a new one, so we don't need to change the ApplyConfigMap API. Probably, we need to have a new API for deleting the configMap.

Changing the conf rev. was a kind of workaround we are using to reset the Pod, otherwise, we never see a use case for the force update to happen the configMap for the same content.  If somebody is using the force option, they should make sure that the code path that gets there is not re-entrant across the reconciles, otherwise it may keep updating unnecessarily.

Reply:
Honestly, either way is fine.  They both require an additional API or API parameter that we can eventually remove once we no longer need the resets.  Only app framework will use the `force` parameter, which does not rerun this code every reconcile due to the appContext status.  I would suspect that even if we created a new ConfigMap it would also have to be non-reentrant or the same issue would occur.

Reply:
IMO, having the delete API is a better option for multiple reasons: 
- We are getting a new API :-).  
- We don't need to change the API behavior that is not something for a standard use case. 
- We don't need to address it in several places, which we need to revert again after some time.

Otherwise, I am fine with either option :-).

Reply:
Actually, looking back at it, even the DELETE and CREATE calls are expensive, and we can avoid it.
All that we need is, just one more UPDATE call. i.e calling `splctrl.ApplyConfigMap` once with an empty data map. That should solve our unique situation.

Reply:
Ya I agree that a delete and readd is likely much more expensive than a modify.  So I can basically add a reset of the config prior to the set.  This means if we have any app updates and the appListing changes, we would now update the configmap twice: once setting the `Data` section to "" and a second time to set it to the actual `appListing` as opposed to the original fix that only updated the configmap once.  This way there are no changes to the `ApplyConfigMap()` API and this is completely isolated to the AppFramework code.

Since the two updates are in quick secession, there shouldn't be any adverse side effects.  I'm testing it out to verify this.

Reply:
One way we can avoid extra update calls is by doing a deep equal comparison on the existing configMap data. However, that needs a READ CRUD call. So, I guess, we are fine with this approach, considering we are patching a unique use case. :-)